src/tendencies/edmfx_closures.jl

```diff
@@ -0,0 +1,175 @@
+        # TODO - using Π₁ blows things up
+        #Π₁ = ᶜz * (ᶜbuoyʲ - ᶜbuoy⁰) / ((ᶜwʲ_phys - ᶜw⁰_phys)^2 + w_star^2 + eps(FT))
+        Π₃ = sqrt(ᶜaʲ)
+        Π₄ = ᶜRHʲ - ᶜRH⁰
+        Π₆ = ᶜz / ref_H
+
+        entr = max(
+            0,
+            ᶜwʲ_phys / ᶜz * (
+                -4.013288 - 0.000968 * Π₁ + 0.356974 * Π₃ - 0.403124 * Π₄ +
+                1.503261 * Π₆
+            ),
+        )
+        detr = max(
+            0,
+            ᶜwʲ_phys * (
+                3.535208 + 0.598496 * Π₁ + 1.583348 * Π₃ + 0.046275 * Π₄ -
+                0.344836 * Π₆ + max_area_limiter
+            ),
+        )
+        return (; entr, detr)
+    end
+end
+
+edmfx_entr_detr_cache(Y, turbconv_model) = (;)
+function edmfx_entr_detr_cache(Y, turbconv_model::EDMFX)
+    n = n_mass_flux_subdomains(turbconv_model)
+    FT = eltype(Y)
+    return (;
+        ᶜentr_detrʲs = similar(
+            Y.c,
+            NTuple{n, NamedTuple{(:entr, :detr), NTuple{2, FT}}},
+        )
+    )
+end
+
+edmfx_entr_detr_tendency!(Yₜ, Y, p, t, colidx, turbconv_model) = nothing
+function edmfx_entr_detr_tendency!(Yₜ, Y, p, t, colidx, turbconv_model::EDMFX)
+
+    n = n_mass_flux_subdomains(turbconv_model)
+    ᶜlg = Fields.local_geometry_field(Y.c)
+
+    (; params, ᶜp, ᶜρ_ref, buoy_flux_surface) = p
+    (; ᶜρʲs, ᶜtsʲs, ᶜuʲs, ᶜspecificʲs, ᶜentr_detrʲs) = p
+    (; ᶜρ⁰, ᶜts⁰, ᶜu⁰, ᶜspecific⁰) = p
+
+    thermo_params = CAP.thermodynamics_params(params)
+
+    ᶜz = Fields.coordinate_field(Y.c).z
+
+    for j in 1:n
+        @. ᶜentr_detrʲs.:($$j)[colidx] = pi_groups_entr_detr(
+            params,
+            ᶜz[colidx],
+            ᶜp[colidx],
+            Y.c.ρ[colidx],
+            buoy_flux_surface[colidx],
+            Y.c.sgsʲs.:($$j).ρa[colidx] / ᶜρʲs.:($$j)[colidx],
+            get_physical_w(ᶜuʲs.:($$j)[colidx], ᶜlg[colidx]),
+            TD.relative_humidity(thermo_params, ᶜtsʲs.:($$j)[colidx]),
+            ᶜbuoyancy(params, ᶜρ_ref[colidx], ᶜρʲs.:($$j)[colidx]),
+            get_physical_w(ᶜu⁰[colidx], ᶜlg[colidx]),
+            TD.relative_humidity(thermo_params, ᶜts⁰[colidx]),
+            ᶜbuoyancy(params, ᶜρ_ref[colidx], ᶜρ⁰[colidx]),
+        )
+        @. Yₜ.c.sgsʲs.:($$j).ρa[colidx] +=
+            Y.c.sgsʲs.:($$j).ρa[colidx] * (
+                ᶜentr_detrʲs.:($$j).entr[colidx] -
+                ᶜentr_detrʲs.:($$j).detr[colidx]
+            )
+
+        @. Yₜ.c.sgsʲs.:($$j).ρae_tot[colidx] +=
+            Y.c.sgsʲs.:($$j).ρa[colidx] * (
+                ᶜentr_detrʲs.:($$j).entr[colidx] * TD.total_specific_enthalpy(
+                    thermo_params,
+                    ᶜts⁰[colidx],
+                    ᶜspecific⁰.e_tot[colidx],
+                ) -
+                ᶜentr_detrʲs.:($$j).detr[colidx] * TD.total_specific_enthalpy(
+                    thermo_params,
+                    ᶜtsʲs.:($$j)[colidx],
+                    ᶜspecificʲs.:($$j).e_tot[colidx],
+                ) -
+                ᶜp[colidx] / Y.c.ρ[colidx] * (
```

Comment:
You shouldn't need this term here anymore:)

Reply:
Great!